kfs-cr/src/main/java/edu/arizona/kfs/module/cr/batch/service/impl/CheckReconciliationImportBatchServiceImpl.java

```diff
@@ -0,0 +1,504 @@
+        }
+        writeLog();
+        return success;
+    }
+
+    // Private Methods
+
+    private Map<String, String> getStatusMapFromParameter(String parameterName, String statusCode) {
+        Map<String, String> retval = new HashMap<String, String>();
+        String parameterString = getParameter(parameterName);
+
+        StringTokenizer parser = new StringTokenizer(parameterString, KFSConstants.MULTI_VALUE_SEPERATION_CHARACTER, false);
+        while (parser.hasMoreTokens()) {
+            String temp = parser.nextToken();
+            retval.put(temp, statusCode);
+        }
+
+        return retval;
+    }
+
+    private String getParameter(String parameterName) {
+        String retval = parameterService.getParameterValueAsString(CrParameterConstants.CR_NAMESPACE_CODE, CrParameterConstants.CHECK_RECONCILIATION_IMPORT_STEP_COMPONENT, parameterName);
+        if (StringUtils.isBlank(retval)) {
+            throw new RuntimeException("System parameter " + parameterName + " is null.");
+        }
+        return retval;
+    }
+
+    private Collection<CheckReconciliation> getNewCheckReconciliations() {
+        Collection<CheckReconciliation> retvals = checkReconciliationDao.getNewCheckReconciliations(banks);
+        return retvals;
+    }
+
+    private void writeDoneFile(File file) {
+        String strippedOfName = FilenameUtils.removeExtension(file.getAbsolutePath());
+        String doneFileName = strippedOfName + GeneralLedgerConstants.BatchFileSystem.DONE_FILE_EXTENSION;
+        File doneFile = new File(doneFileName);
+        try {
+            doneFile.createNewFile();
+        } catch (IOException e) {
+            LOG.error("Unable to recreate .done file: " + doneFileName);
+        }
+    }
+
+    private List<File> getDataFileList() {
+        List<File> fileList = new ArrayList<File>();
+        List<File> doneFileList = new ArrayList<File>();
+        File file = new File(stagingDirectory);
+
+        if (file.isDirectory()) {
+            doneFileList = Arrays.asList(file.listFiles(new DoneFilenameFilter()));
+        }
+
+        for (File donefile : doneFileList) {
+            String fileName = donefile.getAbsolutePath();
+            fileName = FilenameUtils.removeExtension(fileName) + GeneralLedgerConstants.BatchFileSystem.EXTENSION;
+            fileList.add(new File(fileName));
+        }
+
+        return fileList;
+    }
+
+    private void deleteDoneFile(File dataFile) {
+        String strippedOfName = FilenameUtils.removeExtension(dataFile.getAbsolutePath());
+        String doneFile = strippedOfName + GeneralLedgerConstants.BatchFileSystem.DONE_FILE_EXTENSION;
+
+        try {
+            FileUtils.deleteQuietly(new File(doneFile));
+        } catch (Exception e) {
+            LOG.info("Error in CheckReconciliationBatchService in method deleteDoneFile");
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Transactional
+    private void processFile(File file) throws Exception {

```

Comment:
Finally and additional try blocks added at potential exception points, closing the BufferedReader before throwing exceptions.